File: packages/ios/FxPressHost.swift
import UIKit

internal protocol FxPressHost: AnyObject {
  func hitTarget(point: CGPoint) -> Bool
  func handlePressBegin(point: CGPoint, depth: Int)
  func handlePressChanged(point: CGPoint, depth: Int)
  func handlePressEnd(point: CGPoint, includePressEvent: Bool)
  func handlePressCancel(point: CGPoint)
  func handleLongPress(point: CGPoint)
  func attachRecognizer(_ recognizer: UILongPressGestureRecognizer)
  func detachRecognizer(_ recognizer: UILongPressGestureRecognizer)
}

File: packages/android/src/main/java/expo/modules/reactnativefx/FxPressHost.kt
package expo.modules.reactnativefx

import android.view.GestureDetector

internal interface FxPressHost {
  fun hitTarget(x: Float, y: Float): Boolean
  fun handlePressBegin(x: Float, y: Float, depth: Int)
  fun handlePressChanged(x: Float, y: Float, depth: Int)
  fun handlePressEnd(x: Float, y: Float, includePressEvent: Boolean)
  fun handlePressCancel(x: Float, y: Float)
  fun handleLongPress(x: Float, y: Float)
  fun attachRecognizer(recognizer: GestureDetector)
  fun detachRecognizer(recognizer: GestureDetector)
}
